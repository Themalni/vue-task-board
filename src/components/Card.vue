<template>
  <div class="card" @click="openCard()">
    <div class="card-body">{{ truncateString(description, 65) }}</div>
    <div class="card-footer">
      <slot></slot>
    </div>
  </div>
</template>

<script>
export default {
  name: "Card",
  props: {
    description: {
      type: String
    }
  },
  methods: {
    openCard() {
      this.$emit("openCard");
    },
    truncateString(text, limit) {
      const shortened = text.indexOf(" ", limit);
      if (shortened == -1) return text;
      return text.substring(0, shortened) + "...";
    }
  }
};
</script>

<style lang="scss">
.card {
  position: relative;
  max-width: 100%;
  height: 70px;
  margin-bottom: 0.7em;
  padding: 0.5em;
  border-radius: 4px;
  background-color: $white;
  box-shadow: 0 1px 5px -3px $graphite;
  font-size: 0.8em;
  overflow: hidden;

  &:hover {
    cursor: pointer;
  }
}
.card-footer {
  position: absolute;
  bottom: 0.7em;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
</style>
