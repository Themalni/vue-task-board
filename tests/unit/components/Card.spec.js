import { mount } from '@vue/test-utils'
import Card from '@/components/Card'

describe('Card', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(Card, {
      propsData: {
        description: ''
      }
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('does contain `card` element', () => {
    expect(wrapper.classes('card')).toBe(true)
  })
  it('calls `truncateString` and returns shortened string', () => {
    const text = 'Some long text to shorten for testing purposes'
    wrapper.vm.truncateString(text, 14)

    expect(wrapper.vm.truncateString(text, 14)).toBe('Some long text...')
  })
  it('receives text when value is provided to `description` prop', () => {
    const propsData = {
      description: 'Test description'
    }
    wrapper.setProps(propsData)
    const taskDescription = wrapper.find('.card-body')

    expect(taskDescription.text()).toBe('Test description')
  })
  it('emits event `openCard`', () => {
    wrapper.vm.openCard()

    expect(wrapper.emitted().openCard).toBeTruthy()
    expect(wrapper.emitted().openCard.length).toBe(1)
  })
})
