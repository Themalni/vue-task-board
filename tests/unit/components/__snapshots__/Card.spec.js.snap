// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div class="card">
  <div class="card-body"></div>
  <div class="card-footer"></div>
</div>
`;
